CMakeLists.txt

```diff
@@ -13,13 +13,18 @@
 # limitations under the License
 
 cmake_minimum_required(VERSION 3.0)
+
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-set(PROJ_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
-set(PROJ_BINARY_ROOT ${CMAKE_CURRENT_BINARY_DIR})
 
 include(system)
 
-project(paddle CXX C Go)
+project(paddle VERSION 0.10.0 LANGUAGES CXX C Go)
```

Comment:
I am afraid that the version written here could be inconsistent with the Git tag.  Can we still derived it from the Git commands but in a more consistent way?

My primary concern with the current Git-derived version number is that it considers untagged Git commits the nearest tagged version.  So that if users run an untagged version and report errors, they cannot tell the exact version they were using.